source/loader/layers/sanitizer/asan_statistics.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ * See LICENSE.TXT
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ * @file asan_statistics.cpp
+ *
+ */
+
+#include "asan_statistics.hpp"
+#include "asan_interceptor.hpp"
+#include "ur_sanitizer_layer.hpp"
+
+namespace ur_sanitizer_layer {
+
+void AsanStats::Print(ur_context_handle_t Context) {
+    getContext()->logger.always("Stats: Context {}", (void *)Context);
+    getContext()->logger.always("Stats:   peak memory overhead: {}%",
+                                Overhead * 100);
```

Comment:
When print, can we also print the detailed data instead of only printing a simple overhead factor?

Reply:
I didn't save detailed data. In the end, all saved data should be 0 if no memory leak.